client/app/widgets/QuickCreate/index.jsx

```diff
@@ -71,30 +71,33 @@ export class QuickCreate extends React.Component<Props, State> {
     ).length;
   };
 
-  onCreate = (checkbox: { label: string, value: number, id: string }) => {
-    const { label, value, id } = checkbox;
-    this.setState((prevState: State) => {
-      const { checkboxes } = prevState;
-      checkboxes.push({
-        id,
-        label,
-        value,
-        checked: true,
+  onCreate = (response: any) => {
+    const { data } = response;
+    if (data && data.success) {
+      const { name, id, slug } = data;
+      this.setState((prevState: State) => {
+        const { checkboxes } = prevState;
+        checkboxes.push({
```

Comment:
Let's make this an immutable Array.push
See https://vincent.billey.me/pure-javascript-immutable-array/